### ISSUE TITLE
call of overloaded ‘abs(long int)’ is ambiguous

### ISSUE THREAD

**Issue by faissaloo** [enigma-dev/enigma-dev#958]
```
/home/faissaloo/.enigma/Preprocessor_Environment_Editable/IDE_EDIT_objectfunctionality.h:80:47: error: call of overloaded ‘abs(long int)’ is ambiguous
       score -= abs(fallingbits ^ recievingbits)+ 1;
```

**Comment by faissaloo** [comment]
I did some more testing, the following works:
```
abs(variable)
abs(value)
```
the following doesn't
```
abs(<any expression other than a single variable or value>)
```

**Comment by JoshDreamland** [comment]
Most expressions will work. Your expression is failing because bitwise operators on `var`/`variant` return long. Unfortunately, the best way of dealing with this is probably to either change that fact, or create overloads for `abs(long)`, and any other math functions exhibiting this problem. Looking over them, it seems someone replaced them so that their only overload is `double`, which would be fine if that were absolutely the only overload available. However, that's just not the case, with abs.

**Comment by faissaloo** [comment]
abs(string_length("adfls")) and alot of other functions cause the same issue.

**Comment by RobertBColton** [comment]
Just to mention, this issue is somewhat similar to #214 but I do not consider it a duplicate because it will likely require a separate or additional fix on top of the other one. These variadic/math functions seem to have a lot of issues.

**Comment by fundies** [comment]
tested abs(long) to be working abs(string_length("adfls")) -> abs(size_t) will error because that's pointless.

**Comment by RobertBColton** [comment]
Nope, still an issue, I can reproduce on Win32:

```bash
C:/Users/Owner/AppData/Local/ENIGMA/Preprocessor_Environment_Editable/IDE_EDIT_objectfunctionality.h:42:45: error: call of overloaded 'abs(long int)' is ambiguous
     score -= abs(fallingbits ^ recievingbits)+ 1;
                                             ^
C:/Users/Owner/AppData/Local/ENIGMA/Preprocessor_Environment_Editable/IDE_EDIT_objectfunctionality.h:42:45: note: candidates are:
In file included from c:\users\owner\desktop\enigma\mingw32\include\c++\4.8.2\cstdlib:72:0,
                 from SHELLmain.cpp:20:
c:\users\owner\desktop\enigma\mingw32\i686-w64-mingw32\include\stdlib.h:329:15: note: int abs(int)
   int __cdecl abs(int _X);
               ^
In file included from SHELLmain.cpp:37:0:
Universal_System/mathnc.h:60:20: note: ma_scalar enigma_user::abs(double)
   inline ma_scalar abs(double x)   { return ::fabs(x); }
                    ^
Universal_System/mathnc.h:54:13: note: ma_scalar enigma_user::abs(const var&)
   ma_scalar abs(const var& x);
             ^
Universal_System/mathnc.h:53:13: note: ma_scalar enigma_user::abs(const variant&)
   ma_scalar abs(const variant& x);
             ^
In file included from SHELLmain.cpp:110:0:
C:/Users/Owner/AppData/Local/ENIGMA/Preprocessor_Environment_Editable/IDE_EDIT_objectfunctionality.h:43:45: error: call of overloaded 'abs(long int)' is ambiguous
     score -= abs(fallingbits ^ recievingbits)+ 1;
                                             ^
C:/Users/Owner/AppData/Local/ENIGMA/Preprocessor_Environment_Editable/IDE_EDIT_objectfunctionality.h:43:45: note: candidates are:
In file included from c:\users\owner\desktop\enigma\mingw32\include\c++\4.8.2\cstdlib:72:0,
                 from SHELLmain.cpp:20:
c:\users\owner\desktop\enigma\mingw32\i686-w64-mingw32\include\stdlib.h:329:15: note: int abs(int)
   int __cdecl abs(int _X);
               ^
In file included from SHELLmain.cpp:37:0:
Universal_System/mathnc.h:60:20: note: ma_scalar enigma_user::abs(double)
   inline ma_scalar abs(double x)   { return ::fabs(x); }
                    ^
Universal_System/mathnc.h:54:13: note: ma_scalar enigma_user::abs(const var&)
   ma_scalar abs(const var& x);
             ^
Universal_System/mathnc.h:53:13: note: ma_scalar enigma_user::abs(const variant&)
   ma_scalar abs(const variant& x);
             ^
```

**Comment by RemoveRusky** [comment]
fixed in #1172

**Comment by RobertBColton** [comment]
Double confirm, but I think you meant #1181
